server/src/main/java/io/crate/protocols/postgres/KeyData.java

```diff
@@ -35,4 +40,16 @@ static KeyData generate(int pid) {
     static KeyData of(ByteBuf buffer) {
         return new KeyData(buffer.readInt(), buffer.readInt());
     }
+
+    public static KeyData of(StreamInput in) throws IOException {
+        int pid = in.readInt();
```

Comment:
minor: usually we use `readVInt()` which to my understanding is basically same thing but saving some bytes when possible.

Reply:
It only saves bytes if the values are never negative.  The secret key is random and can get negative.  The pid could afaik overflow and also go negative, although a node would have to stay up a long time for that to happen. Not sure if this matters here.  Cancel shouldn't happen that often

Reply:
thx for explaining, so if it's common to have negative values `VInt` will use 5 bytes instead of 4 and we might end up using more bytes rather than less?